herzog/__init__.py

```diff
@@ -38,10 +39,39 @@ def __exit__(self, *args, **kwargs):
                     del __main__.__dict__[key]
             __main__.__dict__.update(self._state_modules)
 
-def generate(handle: TextIO):
-    cells = [obj.to_ipynb_cell() for obj in parse_cells(handle)
+def load_ipynb_cells(ipynb: TextIO) -> List[Dict[Any, Any]]:
+    try:
+        return json.loads(ipynb.read())['cells']
+    except (json.JSONDecodeError, KeyError):
+        print(f'Check that "{ipynb}" is a valid ipynb file.', file=sys.stderr)
+        raise
+
+def translate_to_ipynb(herzog_handle: TextIO) -> Dict[str, Any]:
+    cells = [obj.to_ipynb_cell() for obj in parse_cells(herzog_handle)
              if obj.has_ipynb_representation]
     with open(os.path.join(os.path.dirname(__file__), "data", "python_3_boiler.json")) as fh:
         boiler = json.loads(fh.read())
-    ipynb = dict(cells=cells, **boiler)
-    return ipynb
+    return dict(cells=cells, **boiler)
+
+def translate_to_herzog(ipynb_handle: TextIO, indent: int = 4) -> Iterable[str]:
+    cells = load_ipynb_cells(ipynb_handle)
+    prefix = ' ' * indent
+    yield 'import herzog\n\n'
+
+    for cell in cells:
+        if isinstance(cell.get('source', None), list):
+            cell['source'] = ''.join(cell['source'])
+
+        if cell['cell_type'] == 'markdown':
+            yield "\nwith herzog.Cell('markdown'):"
+            yield '\n    """\n'
+            if cell['source'].startswith('%') or cell['source'].startswith('!'):
+                yield textwrap.indent('#' + cell['source'], prefix=prefix).rstrip()
```

Comment:
Please use the constants Herzog [defines](https://github.com/xbrianh/herzog/blob/5c16506487943bc7f0b8c642bfa143c3ed716a85/herzog/parser.py#L11-L12) for magic prefixes.

e.g.
```
if cell[‘source’].startswith(“%”):
    source = cell[‘source’].replace(“#”, parser.JUPYTER_MAGIC_PFX, 1)
```